aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -305,9 +321,12 @@ private Completable startApiSync() {
 
     private void stopApiSyncBlocking() {
         try {
-            stopApiSync()
+            boolean subscribed = stopApiSync()
                 .subscribeOn(startStopScheduler)
-                .blockingAwait(OP_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+                .blockingAwait(adjustedTimeoutSeconds, TimeUnit.SECONDS);
```

Comment:
Here, the timeout is being applied "outside of" `stopApiSync()`. For `startApiSync()`, though, the timeout is applied _inside_, on the `syncProcessor.hydrate()`. Is there any way to change this, so that the blocking timeout happens at the same abstraction level, on start/stop APIs, as opposed to on stop, and on a detail of start?

Reply:
For the start scenario, we also have to take into account that we block inside the `SubscriptionProcessor` as well as on the call to `syncProcessor.hydrate()` as you pointed out. So if we move the timeout higher in the call hierarchy (for start), we'll have to make sure that timeout is doubled since the worst case scenario would be for both SubscriptionProcessor and SyncProcessor to take `numberOfModels* TIMEOUT_SECONDS_PER_MODEL` seconds

Reply:
I think this effort is unnecessary. Correct me if i'm wrong, but shouldn't `blockingAwait()` be applied to any operation where you want to ensure that it is completed before moving onto the next instruction? What is the purpose of putting a blocking await on the entire `startApiSync()` method instead of on its components that actually need it?